consensus/blake3pow/consensus.go

```diff
@@ -450,20 +450,31 @@ func (blake3pow *Blake3pow) verifyHeader(chain consensus.ChainHeaderReader, head
 		if header.StateLimit() != expectedStateLimit {
 			return fmt.Errorf("invalid stateLimit: have %v, want %v, parentStateLimit %v", expectedStateLimit, header.StateLimit(), parent.StateLimit())
 		}
+		expectedBaseFee := chain.CalcBaseFee(parent)
+		if header.BaseFee().Cmp(expectedBaseFee) != 0 {
+			return fmt.Errorf("invalid baseFee: have %v, want %v, parentBaseFee %v", expectedBaseFee, header.BaseFee(), parent.BaseFee())
+		}
 		var expectedPrimeTerminusHash common.Hash
+		var expectedPrimeTerminusNumber *big.Int
 		_, parentOrder, _ := blake3pow.CalcOrder(chain, parent)
 		if parentOrder == common.PRIME_CTX {
 			expectedPrimeTerminusHash = parent.Hash()
+			expectedPrimeTerminusNumber = parent.Number(common.PRIME_CTX)
 		} else {
 			if chain.IsGenesisHash(parent.Hash()) {
 				expectedPrimeTerminusHash = parent.Hash()
+				expectedPrimeTerminusNumber = parent.Number(common.PRIME_CTX)
 			} else {
 				expectedPrimeTerminusHash = parent.PrimeTerminusHash()
+				expectedPrimeTerminusNumber = parent.PrimeTerminusNumber()
 			}
 		}
 		if header.PrimeTerminusHash() != expectedPrimeTerminusHash {
 			return fmt.Errorf("invalid primeTerminusHash: have %v, want %v", header.PrimeTerminusHash(), expectedPrimeTerminusHash)
 		}
+		if header.PrimeTerminusNumber().Cmp(expectedPrimeTerminusNumber) != 0 {
```

Comment:
Is it possible for the hash to be the same but the number to be different?

Reply:
its not possible because of this exact validation, a block with same hash has to have same number

Reply:
So this is just a sanity check right? Because even without the check it's impossible